pkg/controller/storagecluster/storagecluster.go

```diff
@@ -429,31 +430,24 @@ func (c *Controller) manage(
 	)
 	zoneMap := make(map[string]int)
 
-	// Get the cloud provider
+	cloudProvider := cloudprovider.New(cloudProviderName)
```

Comment:
cloudProviderName here is nil or empty.
The cloudProviderName gets populated in the next for loop. Once that gets generated then call 
cloudprovider.New(cloudProviderName)